flow/connectors/postgres/postgres.go

```diff
@@ -93,18 +83,36 @@ func NewPostgresConnector(ctx context.Context, pgConfig *protos.PostgresConfig,
 		ctx:                ctx,
 		config:             pgConfig,
 		pool:               pool,
-		replPool:           replPool,
+		replConfig:         replConfig,
+		replPool:           nil,
 		customTypesMapping: customTypeMap,
 		metadataSchema:     metadataSchema,
 		logger:             *flowLog,
 	}, nil
 }
 
-// GetPool returns the connection pool.
+// nil returns the connection pool.
 func (c *PostgresConnector) GetPool() *SSHWrappedPostgresPool {
 	return c.pool
 }
 
+func (c *PostgresConnector) GetReplPool(ctx context.Context) (*SSHWrappedPostgresPool, error) {
+	c.replMutex.Lock()
```

Comment:
why do we need a mutex? afaik connectors are not shared across mirrors and I don't see a way this can be invoked simultaneously by the same mirror either.

Reply:
I didn't know if connectors needed to be threadsafe, so I assumed yes. If we're fine explicitly declaring that connectors shouldn't be used concurrently across goroutines or anything this can be pared down